components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java

```diff
@@ -107,28 +108,42 @@ public void create(IdentityRequest.IdentityRequestBuilder builder, HttpServletRe
                 log.debug("DCR request json : " + jsonData.toJSONString());
             }
 
-            RegistrationRequestProfile registrationRequestProfile =
-                    registerRequestBuilder.getRegistrationRequestProfile();
+            RegistrationRequestProfile registrationRequestProfile = registerRequestBuilder
+                    .getRegistrationRequestProfile();
             if (registrationRequestProfile == null) {
                 registrationRequestProfile = new RegistrationRequestProfile();
             }
 
             Object obj = jsonData.get(RegistrationRequest.RegisterRequestConstant.REDIRECT_URIS);
             if (obj != null && obj instanceof JSONArray) {
                 JSONArray redirectUris = (JSONArray) obj;
-                for (int i = 0; i < redirectUris.size(); i++) {
-                    registrationRequestProfile.getRedirectUris().add(redirectUris.get(i).toString());
+                for (Object redirectUri : redirectUris) {
+                    //ToDO invalid_redirect_uri - error code should be sent to client in below error cases.
+
+                    if (redirectUri == null || IdentityUtil.isBlank(redirectUri.toString())) {
+                        throw IdentityException
+                                .error(FrameworkClientException.class, "The redirection URI is either null or blank.");
+                    }
+
+                    String redirectUriStr = redirectUri.toString();
```

Comment:
better to avoid making multiple toString on redirectUri. Also better name for String representation of redirectUri than redirectUriStr. not a must to do